h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java

```diff
@@ -255,6 +254,10 @@ V getFormattedValue() {
     @API(help="The list of modeling steps to be used by the AutoML engine (they may not all get executed, depending on other constraints).",
             level = API.Level.expert)
     public StepDefinitionV99[] modeling_plan;
+    
+    @API(help="The list of preprocessing steps to run. Only 'target_encoding' is currently supported.",
+            level = API.Level.secondary)
+    public PreprocessingStepDefinitionV99[] preprocessing;
```

Comment:
@sebhrusen Is there any special reason for introducing a dedicated class for this enum? GLM uses the enum directly: https://github.com/h2oai/h2o-3/blob/a8a67c3623da87b5cd814c2ee970c6c1132a2049/h2o-algos/src/main/java/hex/schemas/GLMV3.java#L98

I will have to implement an exception for this in Sparkling Water.

Reply:
sorry, your link point to enum `GLMParameters.Family`, I don't see the connection with my changes here.

Reply:
if you mean: why did I create `ProcessingStepDefinitionV99`, it's because I will add more parameters in the future (this is just minimal first integration here) for example to configure TE, and we also have plans for more preprocessors...
So having this class now allows more flexibility and backwards compatibility in the future.

Reply:
@sebhrusen 

> why did I create ProcessingStepDefinitionV99

Yes, exactly. I used the `rand_family` parameter just as an example how it's done elsewhere. (Propagation of an array of enum values from a client to  the h2o backend).

> for example to configure TE
Thanks for giving me the context! It make sense to me. Just one more question... If you add some parameters to  `ProcessingStepDefinitionV99`, can i expect that type of [the `preprocessing` parameter](https://github.com/h2oai/h2o-3/pull/4927/files#diff-9f262b275056f042a5247e16d4bf59c9R27) in Python client will change from a string array to something more complex?

Reply:
correct, clients like Py and R, will support   a simple syntax (string) and an advanced one (dictionary): currently, when user is passing a string, it's actually already translated into a dictionary internally, so that later we will just open the dictionary syntax to the end user.